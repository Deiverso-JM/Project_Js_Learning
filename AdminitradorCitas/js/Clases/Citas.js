



class Citas {
    constructor() {
        this.citas = [];
    }

    agregarCitas(cita) {
        this.citas = [...this.citas, cita];
    }

    eliminarCita(id) {
        this.citas = this.citas.filter((cita) => cita.id !== id);
    }


    editarCita(citaactualizada){

        this.citas = this.citas.map(cita => cita.id === citaactualizada.id ? citaactualizada : cita)

    }
}


export default Citas;