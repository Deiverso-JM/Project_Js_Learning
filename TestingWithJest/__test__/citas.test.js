import Citas from '../js/classes/Citas.js'

describe('Citas testing', () => {
    const citas = new Citas();
    const id = Date.now();
    test('agregar una nueva cita', () => { 
        const citaObj = {
            id,
            mascota: 'Hook',
            propietario: 'Johan',
            telefono: '154564',
            fecha: '10-32-4123',
            hora:'10:30',
            sintomas: 'solo duerme'
        }



        citas.agregarCita(citaObj);

        //Prueba
        expect(citas).toMatchSnapshot();

        
     })

     test('Actualizar Cita', () => { 
        const citaActualizada = {
            id,
            mascota: 'Hook',
            propietario: 'Johan',
            telefono: '154564',
            fecha: '10-32-4123',
            hora:'10:30',
            sintomas: 'solo duerme'
        }


        citas.editarCita(citaActualizada);

        //Prueba
        expect(citas).toMatchSnapshot();

        
     })

     test('Eliminar Cita', () => { 
        const citaEliminada = {
            id,
            mascota: 'Hook',
            propietario: 'Johan',
            telefono: '154564',
            fecha: '10-32-4123',
            hora:'10:30',
            sintomas: 'solo duerme'
        }


        citas.eliminarCita(citaEliminada);

        //Prueba
        expect(citas).toMatchSnapshot();

        
     })
})