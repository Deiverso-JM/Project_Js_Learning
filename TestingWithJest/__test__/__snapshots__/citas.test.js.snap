// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Citas testing Actualizar Cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-32-4123",
      "hora": "10:30",
      "id": 1698637986596,
      "mascota": "Hook",
      "propietario": "Johan",
      "sintomas": "solo duerme",
      "telefono": "154564",
    },
  ],
}
`;

exports[`Citas testing Eliminar Cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-32-4123",
      "hora": "10:30",
      "id": 1698637986596,
      "mascota": "Hook",
      "propietario": "Johan",
      "sintomas": "solo duerme",
      "telefono": "154564",
    },
  ],
}
`;

exports[`Citas testing agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-32-4123",
      "hora": "10:30",
      "id": 1698637986596,
      "mascota": "Hook",
      "propietario": "Johan",
      "sintomas": "solo duerme",
      "telefono": "154564",
    },
  ],
}
`;
